### ISSUE TITLE
Failed test with Python 3.11: property of 'Mock' object has no <...>

### ISSUE THREAD

**Issue by s3v-** [horazont/aioxmpp#385]
Two tests start failing with Python 3.11
See: [https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=1028835](https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=1028835)

```python
=================================== FAILURES ===================================
______________ Testproxy_property.test_rejects_deletes_by_default ______________
AttributeError: property of 'Mock' object has no deleter
 
During handling of the above exception, another exception occurred:
 
self = <tests.test_utils.Testproxy_property testMethod=test_rejects_deletes_by_default>
 
    def test_rejects_deletes_by_default(self):
        self.assertTrue(hasattr(self.obj.member, "attr"))
     
>       with self.assertRaisesRegex(AttributeError, "can't delete attribute"):
E       AssertionError: "can't delete attribute" does not match "property of 'Mock' object has no deleter"
 
tests/test_utils.py:766: AssertionError
______________ Testproxy_property.test_rejects_writes_if_readonly ______________
AttributeError: property of 'Mock' object has no setter

During handling of the above exception, another exception occurred:
 
self = <tests.test_utils.Testproxy_property testMethod=test_rejects_writes_if_readonly>
 
    def test_rejects_writes_if_readonly(self):
        self.obj.member.attr = unittest.mock.sentinel.old_value
     
        pp = utils.proxy_property("member", "attr", readonly=True)
     
>       with self.assertRaisesRegex(AttributeError, "can't set attribute"):
E       AssertionError: "can't set attribute" does not match "property of 'Mock' object has no setter"
 
tests/test_utils.py:776: AssertionError
```

This is due to a change in AttributeError message [https://github.com/python/cpython/pull/31311](https://github.com/python/cpython/pull/31311).

Python 3.11
```python
Python 3.11.1 (main, Dec 31 2022, 10:23:59) [GCC 12.2.0] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> from aioxmpp import utils
>>> pp = utils.proxy_property("member", "attr", readonly=True)
>>> from unittest.mock import Mock
>>> obj = Mock(["member"])
>>> pp.__delete__(obj)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
AttributeError: property of 'Mock' object has no deleter
>>> 
```

Python 3.10 or below:
```python
Python 3.10.9 (main, Dec  7 2022, 13:47:07) [GCC 12.2.0] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> from aioxmpp import utils
>>> pp = utils.proxy_property("member", "attr", readonly=True)
>>> from unittest.mock import Mock
>>> obj = Mock(["member"])
>>> pp.__delete__(obj)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
AttributeError: can't delete attribute
>>> 
```
Kind Regards

**Comment by horazont** [comment]
Thank you for also filing this upstream :-). I'll look into it shortly.

**Comment by horazont** [comment]
Fixed in #386, I also released 0.13.3 and prepared a new version of the Debian package. Thanks a lot!
